src/dump.c

```diff
@@ -2422,6 +2451,7 @@ static jl_method_t *jl_lookup_method(jl_methtable_t *mt, jl_datatype_t *sig, siz
 
 static jl_method_t *jl_recache_method(jl_method_t *m)
 {
+    assert(!m->is_for_opaque_closure);
```

Comment:
Seems like this assert isn't handled on serialization? (there's nothing stopping the user from referencing this from other places via reflection or perhaps `.roots` from generated code)

Reply:
I'll add an error case if you try to serialize it.